lib/provisioner/worker/plugins/automators/chef_solo_automator/chef_solo_automator.rb

```diff
@@ -51,7 +53,9 @@ def generate_chef_primitive_tar(chef_primitive)
     # rubocop:disable GuardClause
     if !File.exist?(chef_primitive_tar) || ((Time.now - File.stat(chef_primitive_tar).mtime).to_i > 600)
       log.debug "Generating #{chef_primitive_tar} from #{chef_primitive_path}"
-      `tar -chzf "#{chef_primitive_tar}.new" #{chef_primitive}`
+      newfile = gzip(tar(chef_primitive_path))
+      File.new("#{chef_primitive_tar}.new", 'w').write newfile.read
```

Comment:
Put `newfile.read` inside parentheses, to make it clear that it's the argument to write.